clientgen/clientgen.gotpl

```diff
@@ -9,16 +9,20 @@ type Client interface {
 	// the proceeding Client method requests with the given string
 	WithQuery(s string) Client
 
-	{{ range $field := .QueryRoot.Fields -}}
-		{{- if $field.IsResolver }}
-			{{- $field.GoFieldName}}{{ $field.ShortResolverDeclaration }}
-		{{- end }}
-	{{ end }}
-	{{ range $field := .MutationRoot.Fields -}}
-		{{- if $field.IsResolver }}
-			{{- $field.GoFieldName}}{{ $field.ShortResolverDeclaration }}
-		{{- end }}
-	{{ end }}
+    {{- if .HasQuery -}}
```

Comment:
I believe you can just do `{{- if .QueryRoot -}}` and you won't need the two methods `HasMutation` and `HasQuery`